### ISSUE TITLE
Custom images for not switch devices not working

### ISSUE THREAD

**Issue by ultrasuperpingu** [domoticz/domoticz#4491]
It seems that custom image in python plugins do not work with devices other than switches. It's probably a limitation, not really a bug but could it be documented in the plugin development documentation ? Better, is it possible to add a log when someone try to change Image of a device that do not support this feature ?

**Comment by gizmocuz** [comment]
Not sure what you mean, but you can only change the icon for switches and custom sensors

**Comment by ultrasuperpingu** [comment]
I didn't know that Custom Devices was supporting Custom Images. It's good to know :)
Is it possible to add this information in the plugin devepment documentation ? It would be useful I think to say in this doc that it is not possible to use Custom Images on other type of Devices.

So, I just wondering, for a single numerical value sensor, it seems it's always better to use Custom Devices , isn't it ? I don't see features in, let's say a Pourcentage sensor that the Custom Sensor wouldn't have ?

For the log, I was talking about logging an error if somebody try to do something like:
`Domoticz.Device(Name="test",` Unit=1, TypeName="Percentage", Image=12).Create()`
or
`dev.Update(0,"12",Image=42)`

**Comment by gizmocuz** [comment]
Depends, we also have counter devices , depends on the usage
I have no idea what that call is with image=12 or image=42, is this dzVents ?
Then better open a topic on the forum

**Comment by ultrasuperpingu** [comment]
sorry, it's in python plugin but never mind, the task is probably just to update documentation.
Thanks

**Comment by gizmocuz** [comment]
The python documentation maybe.
Best to open a topic on the Forum for this (Under the Python section)

**Comment by ultrasuperpingu** [comment]
Is there a particular reason for this limitation (Custom Images only for Switches and Custom Sensors) ? I just look at the code and there is a huge if in WebServer.cpp to add an Image field only on those devices. If we put the "look if the device have a Custom Image and add Image field if it does" outside this if, javascript side would have access to the filename. It's pretty easy to do and (maybe I'm wrong) it seems pretty safe to do that. After, there is some work on javascript side to test the existence of an Image field and use it but I think it is not that much work. I'd like to try. Do you see any no-go ?

**Comment by gizmocuz** [comment]
I'm not against it, sure you can try. But for me personally I think the current icons are sufficient and OK.
Why people always want to change the looks is beyond a reason I can think of. You also can't change the layout of your Router/NAS/Alexa/Hue/Google Home App

**Comment by ultrasuperpingu** [comment]
I agree, current icons are ok and probably sufficient in most case. But when you develop a plugin for a specific device and use for it an existing device type, I think it is natural to want icons of this device to reflect the real device look (but I agree, it is not necessary). 

For me, it's just that while reading the documentation, I thought it was possible, so I made those icons (I work with graphists who tell me that polishing look is important, that's probably why 😄), and I finally found out it was not. That's why I'd like to be able to use them :)

Now, if you think it's not a good idea (because the code would be more complicated, or for any other reason), no problem, just tell me. If not, I'll try to implement it when I'll have some spare time.

For this issue, maybe I should change the title to something like: "Add possibility to use Custom Images on non switch/custom devices" and tag it as enhancement ?
